### DISCUSSION TITLE
ARexx port

### DISCUSSION THREAD

**Discussion by Dobrowit** [sacredbanana/AmigaGPT#7] [Ideas]
It would be nice to have a port for ARexx scripters in this application. The set of commands should allow communication and data transfer between other applications.

For example:
- when writing something in a text editor (e.g. CED), you could send a fragment of the text to be translated and paste back it automatically
- automatically send generated graphics to graphic programs
- calling the ChatGPT function from other applications, e.g. in TVPaint, the script after entering the command would generate the appropriate graphic or brush
e.t.c.

In addition, it would be useful to add an additional menu in AmigaGPT itself for connecting your own scripts - as e.g. FileX has.

**Comment by sacredbanana** [comment]
Thanks for the idea! I have added it as an issue #11 enhancement for a future release. I'll have to do some learning about ARexx :D

**Comment by sacredbanana** [comment]
@Dobrowit Just to let you know, this is still very much something I plan to look into. I have not forgotten about this. It's a great idea with a lot of potential

**Reply by Dobrowit** [reply]
Potentially, any ancient application can be enriched with AI functionality in this way.

Now I came up with something like this - in DOpus you select a text file, the script sends this file to AI to generate a sensible file title or comment based on the content. In any text editor, you could implement text translation into any language, summarizing, correcting, etc. Maybe integration with some editor for programmers.

**Reply by sacredbanana** [reply]
Good news! I have just added ARexx support in 2.4.0! I have made a few ARexx commands available for people to use in their scripts. See the docs for more info and an example script

**Reply by sacredbanana** [reply]
Sorry you had to wait over a year for this. I had a lot of stuff I wanted to add first and I am only one person.

Also I plan to later on have a menu option in the app for running your own ARexx macros

**Comment by sacredbanana** [comment]
Feature added.
